relay-general/src/pii/compiledconfig.rs

```diff
@@ -38,6 +39,7 @@ fn get_rule(config: &PiiConfig, id: &str) -> Option<RuleRef> {
     }
 }
 
+#[allow(clippy::mutable_key_type)]
```

Comment:
This seems to be a false-positive, I'm assuming this is due to `Regexes` stored inside `RuleRef`.